includes/sanitizers/class-amp-base-sanitizer.php

```diff
@@ -603,6 +603,23 @@ public function prepare_validation_error( array $error = [], array $data = [] )
 				( 'style' === $node->nodeName && ! $node->hasAttribute( 'amp-custom' ) && ! $node->hasAttribute( 'amp-keyframes' ) )
 			) {
 				$error['text'] = $node->textContent;
+
+				// Normalize string and number literals to prevent nonces, random numbers, and timestamps from generating endless number of validation errors.
+				$error['text'] = preg_replace(
+					[
+						'/"([^"\n\\\\]|\\\\(\\\\\\\\)*+.)*"/s',
+						'/\'([^\'\n\\\\]|\\\\(\\\\\\\\)*+.)*\'/s',
```

Comment:
Could you explain what the `(\\\\\\\\)` capture group is for?

Reply:
This pattern, `\\\\(\\\\\\\\)*+`, is to match an odd number of backslashes. It's unsightly because the backslashes are being doubly-escaped, but without escaping it would be this pattern:`\(\\)*+`.

Reply:
Oh OK, yea that clears it up :+1:.